File: Q43/Q43.py
#Tower Of Hanoi 
n = int(input("Enter the number of disks: "))

rods = ['A', 'B', 'C']

moves = (1 << n) - 1  

print(f"The sequence of moves to solve the Tower of Hanoi with {n} disks:")

for i in range(1, moves + 1):
    s
    source = (i & (i - 1)) % 3
    destination = ((i | (i - 1)) + 1) % 3

    print(f"Move disk from {rods[source]} to {rods[destination]}")

File: Q43/Q43.cpp
/*Tower Of Hanoi*/
#include <iostream>
#include <cmath>
using namespace std;

int main() 
{
    int n; 

    cout << "Enter the number of disks: ";
    cin >> n;

    int moves = pow(2, n) - 1; 
    char rods[] = {'A', 'B', 'C'}; 

    cout << "The sequence of moves to solve the Tower of Hanoi with " << n << " disks:\n";

    
    for (int i = 1; i <= moves; i++) 
    {
    
        int source = (i & (i - 1)) % 3; 
        int destination = ((i | (i - 1)) + 1) % 3; 

        cout << "Move disk from " << rods[source] << " to " << rods[destination] << endl;
    }

    return 0;
}
